prost-derive/src/field/scalar.rs

```diff
@@ -360,19 +375,38 @@ impl Field {
                 quote!(::core::option::Option::Some(ref val) => &val[..],)
             };
 
-            let get_doc = format!(
+            let match_some2 = if self.ty.is_numeric() {
+                quote!(::core::option::Option::Some(val) => Some(val),)
+            } else {
+                quote!(::core::option::Option::Some(ref val) => Some(&val[..]),)
+            };
+
+            let get_doc_fallback = format!(
                 "Returns the value of `{0}`, or the default value if `{0}` is unset.",
                 ident_str,
             );
+            let get_doc = format!(
+                "Returns the value of `{0}`, or None if `{0}` is unset.",
+                ident_str,
+            );
+            let get_fallback = Ident::new(&format!("{}_fallback", ident_str), Span::call_site());
 
             Some(quote! {
-                #[doc=#get_doc]
-                pub fn #get(&self) -> #ty {
+                #[doc=#get_doc_fallback]
+                pub fn #get_fallback(&self) -> #ty {
                     match self.#ident {
                         #match_some
                         ::core::option::Option::None => #default,
                     }
                 }
+
+                #[doc=#get_doc]
+                pub fn #get(&self) -> ::core::option::Option<#ty> {
```

Comment:
I think this should only return an `Option<T>` for optional protobuf fields. I think it should remain the original code for non-optional fields